src/main/scala/com/sksamuel/scapegoat/sbt/ScapegoatSbtPlugin.scala

```diff
@@ -43,6 +43,8 @@ object ScapegoatSbtPlugin extends AutoPlugin {
       Defaults.compileSettings ++
       Seq(
         sources := (sources in Compile).value,
```

Comment:
My understanding is the same, that extending Compile should cause the task / setting lookup to propagate up the hierarchy. It might be because the scapegoat task is not scoped to the scapegoat config or something.